app/controllers/agent_connect_controller.rb

```diff
@@ -0,0 +1,59 @@
+class AgentConnectController < ApplicationController
+  before_action :log_params_to_sentry
+
+  def auth
+    auth_client = AgentConnectOpenIdClient::Auth.new(login_hint: params[:login_hint])
+    session[:agent_connect_state] = auth_client.state
+    session[:nonce] = auth_client.nonce
+
+    redirect_to auth_client.redirect_url(agent_connect_callback_url), allow_other_host: true
```

Comment:
tu pourrais aussi appeler directement `Rails.application.routes.url_helpers.agent_connect_callback_url` depuis le PORO `Auth` il me semble.

Reply:
oui, mais pour faire ça il faudra passer le `current_domain` depuis le controller, car si on appelle `Rails.application.routes.url_helpers.agent_connect_callback_url` sans argument on aura un `Missing host to link to! Please provide the :host parameter, set default_url_options[:host], or set :only_path to true (ArgumentError)`.
Donc quitte à passer un argument, j'ai une légère préférence pour passer explicitement toute l'url depuis le controller, plutôt que juste un host qui sert ensuite à générer l'url (et c'est pas absurde en terme de responsabilités que le controller connaisse l'url de callback, vu que c'est une action de ce controller)/